stdlib/public/core/Optional.swift

```diff
@@ -299,13 +299,19 @@ public // COMPILER_INTRINSIC
 func _diagnoseUnexpectedNilOptional(_filenameStart: Builtin.RawPointer,
                                     _filenameLength: Builtin.Word,
                                     _filenameIsASCII: Builtin.Int1,
-                                    _line: Builtin.Word) {
-  _preconditionFailure(
-    "unexpectedly found nil while unwrapping an Optional value",
-    file: StaticString(_start: _filenameStart,
-                       utf8CodeUnitCount: _filenameLength,
-                       isASCII: _filenameIsASCII),
```

Comment:
You should use the underscored flavor of `_preconditionFailure` from within the standard library, to avoid circular dependencies when these failures happen within the implementation of `String`. `_preconditionFailure` also handles dropping the location info in non-debug modes, which as @jrose-apple points out is important for secrecy, and I'm inclined to agree with him that type names are also potentially sensitive.